### PR TITLE
Add lib for disassembler operations

### PR DESCRIPTION
**Pull request by johnor**
* Move functions from nestest to new lib.
* Allows a gui to access these functions.

### CODE REVIEW

**Comment by johnor** [comment]
What do you think about this? Makes it easy in a gui to just call disassemble(pc, nes) and get a string that can be used.
We can rename it or the namespace if you have a better suggestion :)

**Comment by robinlinden** [comment]
Looks good! And it living in a separate lib is probably a good idea as not everyone will be interested in disassembling instructions. :P

**Comment by johnor** [comment]
Example from gui:
image

**Comment by johnor** [comment]
Ready now. Will probably cause some conflicts with you refactor of the tests @robinlinden as I had to create a library for test utils. Can wait with this until you have merged..

File: core/test_utils/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,34 @@
+include(FetchContent)
+include(GoogleTest)
+
+project(n_e_s_core_testutils)
+
+FetchContent_Declare(
```
Comment (johnor) [inline core/test_utils/CMakeLists.txt:6]: Should we put all third party dependencies is a new folder by the way and only declare them at one place?

Comment (robinlinden) [reply core/test_utils/CMakeLists.txt:6]: Yes, please. 😁 This was fine back when we only needed the dependency in 1 lib, but it's starting to get a bit messy now.
`third_party/libname`?

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,14 +2,16 @@ cmake_minimum_required(VERSION 3.14)
 
 project(n_e_s)
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
-include(sanitizers)
+include(cmake/sanitizers.cmake)
+include(third_party/gtest.cmake)
```
Comment (johnor) [inline CMakeLists.txt:6]: Like this, or one CMakeLists.txt in third_party with everything?

Comment (robinlinden) [reply CMakeLists.txt:6]: I was thinking more something like
```
third_party/gtest/CMakeLists.txt
third_party/fmt/CMakeLists.txt
```
but this is fine too!

Comment (johnor) [reply CMakeLists.txt]: Sure, I fixed that now :)

File: CMakeLists.txt [hunk 3]
```diff
@@ -2,14 +2,15 @@ cmake_minimum_required(VERSION 3.14)
 
 project(n_e_s)
 
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
```
Comment (robinlinden) [inline CMakeLists.txt:5]: Good change! :D

File: core/BUILD.bazel [hunk 4]
```diff
@@ -11,6 +11,19 @@ cc_library(
     visibility = ["//visibility:public"],
 )
 
+cc_library(
+    name = "core_testutils",
```
Comment (robinlinden) [inline core/BUILD.bazel:15]: Probably change the library name to core_test_utils? Both here an in CMake.

Comment (johnor) [reply core/BUILD.bazel]: Done!

**Pull request merged**